src/components/UriViewer/UriViewer.js

```diff
@@ -56,28 +64,70 @@ export const UriViewer = _.flow(
         setMetadata(metadata)
       }
     } catch (e) {
-      setLoadingError(await e.json())
+      // azure blob storage api only returns responses in xml format at this time
+      // https://feedback.azure.com/d365community/idea/0646f944-3a25-ec11-b6e6-000d3a4f0f84
+      if (isAzureUri(uri)) {
+        setLoadingError(await e.text())
+      } else {
+        setLoadingError(await e.json())
+      }
     }
   }
+
   useOnMount(() => {
     loadMetadata()
   })
 
   const { size, timeCreated, updated, bucket, name, fileName, accessUrl } = metadata || {}
   const gsUri = `gs://${bucket}/${name}`
   const downloadCommand = getDownloadCommand(fileName, gsUri, accessUrl)
```

Comment:
Could we move this code to somewhere where it would only be called if we are in the GCP-context?

It seems a bit wasteful to call it again if we have an Azure URI

Reply:
moved this to be below the Azure modal logic...but that comment you reference is still giving me some confusion on how parts of the modal are rendering for GCP vs Azure that I dont quite understand :/